File: src/main/java/club/deneb/client/features/modules/player/PortalGodMode.java
package club.deneb.client.features.modules.player;

import club.deneb.client.event.events.client.PacketEvent;
import club.deneb.client.features.Category;
import club.deneb.client.features.Module;
import net.minecraft.network.play.client.CPacketConfirmTeleport;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;


/**
 * Created by GlowskiBroski on 10/14/2018.
 */
@Module.Info(name = "PortalGodMode", category = Category.PLAYER)
public class PortalGodMode extends Module {
    @SubscribeEvent
    public void onPacket(PacketEvent.Send event) {
        if (isEnabled() && event.getPacket() instanceof CPacketConfirmTeleport) {
            event.cancel();
        }
    }
}

File: src/main/java/club/deneb/client/features/modules/player/PortalGodMode.kt
package club.deneb.client.features.modules.player

import club.deneb.client.event.events.client.PacketEvent
import club.deneb.client.features.Category
import club.deneb.client.features.Module
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
import net.minecraft.network.play.client.CPacketConfirmTeleport

@Module.Info(name = "PortalGodMode", category = Category.PLAYER)
class PortalGodMode : Module() {
    @SubscribeEvent
    fun onPacket(event: PacketEvent.Send) {
        if (isEnabled && event.getPacket() is CPacketConfirmTeleport) {
            event.cancel()
        }
    }
}